kornia/geometry/transform/pyramid.py

```diff
@@ -329,24 +327,88 @@ def build_pyramid(
         - Input: :math:`(B, C, H, W)`
         - Output :math:`[(B, C, H, W), (B, C, H/2, W/2), ...]`
     """
-    if not isinstance(input, torch.Tensor):
-        raise TypeError(f"Input type is not a torch.Tensor. Got {type(input)}")
-
-    if not len(input.shape) == 4:
-        raise ValueError(f"Invalid input shape, we expect BxCxHxW. Got: {input.shape}")
-
-    if not isinstance(max_level, int) or max_level < 0:
-        raise ValueError(f"Invalid max_level, it must be a positive integer. Got: {max_level}")
+    KORNIA_CHECK_IS_TENSOR(input)
+    KORNIA_CHECK_SHAPE(input, ["B", "C", "H", "W"])
+    KORNIA_CHECK(
+        isinstance(max_level, int) or max_level < 0,
+        f"Invalid max_level, it must be a positive integer. Got: {max_level}",
+    )
 
     # create empty list and append the original image
-    pyramid: List[torch.Tensor] = []
+    pyramid: List[Tensor] = []
     pyramid.append(input)
 
     # iterate and downsample
-
     for _ in range(max_level - 1):
-        img_curr: torch.Tensor = pyramid[-1]
-        img_down: torch.Tensor = pyrdown(img_curr, border_type, align_corners)
+        img_curr: Tensor = pyramid[-1]
+        img_down: Tensor = pyrdown(img_curr, border_type, align_corners)
         pyramid.append(img_down)
 
     return pyramid
+
+
+def is_powerof_two(x):
+    # check if number x is a power of two
+    return x and (not (x & (x - 1)))
+
+
+def find_next_powerof_two(x):
+    # return the nearest power of 2
+    n = math.ceil(math.log(x) / math.log(2))
+    return 2**n
+
+
+def build_laplacian_pyramid(
+    input: Tensor, max_level: int, border_type: str = 'reflect', align_corners: bool = False
+) -> List[Tensor]:
+    r"""Construct the Laplacian pyramid for a tensor image.
+
+    The function constructs a vector of images and builds the Laplacian pyramid
+    by recursively computing the difference after applying
+    pyrUp to the adjacent layer in it's Gaussian pyramid.
+
+    See :cite:`burt1987laplacian` for more details.
+
+    Args:
+        input : the tensor to be used to construct the pyramid.
+        max_level: 0-based index of the last (the smallest) pyramid layer.
+          It must be non-negative.
+        border_type: the padding mode to be applied before convolving.
+          The expected modes are: ``'constant'``, ``'reflect'``,
+          ``'replicate'`` or ``'circular'``.
+        align_corners: interpolation flag.
+
+    Shape:
+        - Input: :math:`(B, C, H, W)`
+        - Output :math:`[(B, C, H, W), (B, C, H/2, W/2), ...]`
+    """
+
+    KORNIA_CHECK_IS_TENSOR(input)
```

Comment:
```suggestion
```

Reply:
You want this check removed?

Reply:
yeah, the `KORNIA_CHECK_SHAPE` already contains that